src/error.rs

```diff
@@ -86,9 +86,13 @@ impl ResponseError for UrlencodedError {
 #[derive(Debug, Display, Error)]
 #[non_exhaustive]
 pub enum JsonPayloadError {
-    /// Payload size is bigger than allowed. (default: 32kB)
-    #[display(fmt = "Json payload size is bigger than allowed")]
-    Overflow,
+    /// Payload size is bigger than allowed. (default: 2MB)
+    #[display(
+        fmt = "JSON payload ({} bytes) is larger than allowed (limit: {} bytes).",
+        size,
+        limit
```

Comment:
I suspect trying to report the size would be confusing because when read chunk-by-chunk it's going to get triggered as soon as the limit is exceeded. We don't necessary know the final payload size.

Reply:
Yep, so the behaviour here is to show the content length header value if present, if not it will be the chunk size that tilts it over the edge.

I tried to make this field an optional but had issues trying to format the error in a nice way, since we don't impose a requirement on content length being present.

Reply:
I've just broken out the enums so that there are two scenarios, one where there is a known length (via the content length header) and one when there isn't.